### PR TITLE
Spicetify Creator refactor

### PR DESCRIPTION
**Pull request by theRealPadster**
Still need to update the install scripts, and the actions.

### CODE REVIEW

**Comment by theRealPadster** [comment]
@afonsojramos You use Windows, right? Could you take a look at or test the install script?

File: resources/install.ps1 [hunk 1]
```diff
@@ -24,18 +24,18 @@ if (-not (Test-Path $sp_dot_dir)) {
...
+Expand-Archive -Path "${HOME}/spicetify-cli/CustomApps/marketplace.zip" -DestinationPath "${HOME}/spicetify-cli/CustomApps/" -Force
+Remove-Item -Path "${HOME}/spicetify-cli/CustomApps/marketplace.zip" -Force
+if (Test-Path -Path "${HOME}/spicetify-cli/CustomApps/marketplace") {
+  Write-Host "marketplace was already found! Updating..." -ForegroundColor "Cyan"
+  Remove-Item -Path "${HOME}/spicetify-cli/CustomApps/marketplace" -Force -Recurse
 }
-Rename-Item -Path "${HOME}/spicetify-cli/CustomApps/spicetify-marketplace-main" -NewName "spicetify-marketplace" -Force
-Copy-Item -Path "${HOME}/spicetify-cli/CustomApps/spicetify-marketplace" -Destination $sp_dot_dir -Recurse -Force
-spicetify config custom_apps spicetify-marketplace
+Rename-Item -Path "${HOME}/spicetify-cli/CustomApps/marketplace-dist" -NewName "marketplace" -Force
```
Comment (kyrie25) [inline resources/install.ps1:36]:
```suggestion
Rename-Item -Path "${HOME}/spicetify-cli/CustomApps/spicetify-marketplace-dist" -NewName "marketplace" -Force
```

**Review by kyrie25 — CHANGES_REQUESTED** [review]
Thrown an error due to wrong folder name
image

File: resources/install.ps1 [hunk 2]
```diff
@@ -24,18 +24,18 @@ if (-not (Test-Path $sp_dot_dir)) {
...
+Copy-Item -Path "${HOME}/spicetify-cli/CustomApps/marketplace" -Destination $sp_dot_dir -Recurse -Force
+spicetify config custom_apps marketplace
```
Comment (kyrie25) [inline resources/install.ps1:38]:
```suggestion
spicetify config custom_apps spicetify-marketplace-
spicetify config custom_apps marketplace
```
Remove the old marketplace from file as it applies both if it is previously installed
image

File: resources/install.ps1 [hunk 3]
```diff
@@ -24,18 +24,18 @@ if (-not (Test-Path $sp_dot_dir)) {
...
 spicetify backup apply
```
Comment (kyrie25) [inline resources/install.ps1:39]:
```suggestion
spicetify backup
spicetify apply
```
Split commands as Spicetify would not let you apply if it is stacked right after `backup`, thus making the user manually applying again.

**Review by kyrie25 — CHANGES_REQUESTED** [review]
Few improvements as well

File: resources/install.ps1 (hunk 3 above)
Comment (CharlieS1103) [reply resources/install.ps1:39]: Are you sure about this? I'm pretty sure it's worked in the past
ex: ```spicetify restore backup apply```

Comment (kyrie25) [reply resources/install.ps1:39]: It works with `spicetify restore backup apply` (the whole chain), not `spicetify backup apply` which would just stop at `backup`

Comment (kyrie25) [reply resources/install.ps1:39]: But using `restore backup apply` would take a significant amount of time plus most of the case Marketplace is installed right after the user installs Spicetify (bundled in docs) which is unnecessary.

Comment (afonsojramos) [reply resources/install.ps1:39]: Maybe this is a bug in spicetify cli, but I've also had the issue detailed above.

**Comment by theRealPadster** [comment]
Thanks @kyrie25! Did I miss anything?

**Comment by kyrie25** [comment]
> Thanks @kyrie25! Did I miss anything?

So far so good yeah!

File: src/components/Grid.tsx [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+      schemes: props.CONFIG.theme.schemes,
+      activeScheme: props.CONFIG.theme.activeScheme,
+      activeThemeKey: props.CONFIG.theme.activeThemeKey,
+    };
+  }
+
+  searchRequested: boolean;
+  endOfList = false;
+  lastScroll = 0;
+  requestQueue: any[] = [];
```
Comment (kyrie25) [inline src/components/Grid.tsx:62]:
```suggestion
  requestQueue: never[][] = [];
```
Not important but if you want to get rid of as most implicit `any` as possible

**Pull request merged**